assembly/JSON.ts

```diff
@@ -183,24 +183,33 @@ export abstract class Value {
 }
 
 export class Str extends Value {
+
   constructor(public _str: string) {
     super();
   }
 
+  private escapeChar(char: string): string {
+    const charCode = char.charCodeAt(0);
+    switch (charCode) {
+      case 0x22: return '\\"';
+      case 0x5C: return "\\\\";
+      case 0x08: return "\\b";
+      case 0x0A: return "\\n";
+      case 0x0D: return "\\r";
+      case 0x09: return "\\t";
+      case 0x0C: return "\\f";
+      case 0x0B: return "\\u000b";
+      default: return char;
+    }
+  }
+
   stringify(): string {
-    let escaped: i32[] = [];
+    let escaped: string[] = [];
```

Comment:
```suggestion
    let escaped: string[] = new Array(this._str.length);
```

Just realized that we should preallocate the array.

Reply:
in such case, we should count the amount of escaped characters, since each of them will add +1 needed space, right? this would in turn translate into two passes into the string. Would that anyway turn into better performance?

Reply:
Hmm good point.  Perhaps we should double the length just for the worst case.  But that only works if the join ignores nulls when generating the string.